src/main/java/com/sweepy/urlService/urlServiceImpl.java

```diff
@@ -0,0 +1,158 @@
+            entry = urlRepository.findByLongUrl(longUrl);
+
+            if (!isValid(protocol + longUrl)) {
+                return "You must enter a valid long Url.";
+            } else if (entry == null || !Objects.equals(entry.getMethod(), method)) {
+                if (Objects.equals(method, "base62")) {
+                    shortUrl = base62Converter.encode(longUrl);
+                } else if (Objects.equals(method, "random")) {
+                    shortUrl = randomConverter.encode(longUrl);
+                } else {
+                    return "Please enter a valid encode method.";
+                }
+                entry = new urlTable(longUrl, shortUrl, method);
+
+                saveToRedis(longUrl, shortUrl, defaultTime);
+                urlRepository.save(entry);
+                return prefix + shortUrl;
+            } else {
+
+                shortUrl = entry.getShortUrl();
+                saveToRedis(longUrl, shortUrl, defaultTime);
+                return prefix + shortUrl;
+            }
+        } else {
+            if (isValid(protocol + longUrl)) {
+                redisService.expire(longUrl, defaultTime);
+                return prefix + shortInRedis;
+            }
+            return "You must enter a valid long Url.";
+
+        }
+
+    }
+
+    @Override
+    @Transactional
+    public String shortToLong(String shortUrl, HttpServletResponse response) throws IOException {
+        urlTable entry;
+        String longUrl;
+
+        String longInRedis = (String) redisService.get(shortUrl);
+
+        if (longInRedis == null) {
+
+
+            try {
+                entry = urlRepository.findByShortUrl(shortUrl);
+                if (shortUrl == "") {
+                    throw new emptyEntry("You must enter a valid short Url.");
+
+                }
+                if (entry == null) {
+                    throw new nullRequest("The short Url: " + shortUrl + " is invalid");
+                }
+                longUrl = protocol + entry.getLongUrl();
+                if (isValid(longUrl)) {
+                    response.sendRedirect(longUrl);
+                    saveToRedis(entry.getLongUrl(), shortUrl, defaultTime);
+                }
+
+            } catch (nullRequest | emptyEntry | IOException e) {
+                return e.getMessage();
+            }
+        } else {
+            longUrl = protocol + longInRedis;
+            if (isValid(longUrl)) {
+                response.sendRedirect(longUrl);
+            } else {
+                return "Please enter a valid url";
+            }
+            redisService.expire(longInRedis, defaultTime);
+        }
+
+
+        return null;
+    }
+
+
+
+    private void saveToRedis(String longUrl, String shortUrl, long time) {
+        System.out.println("outside");
+        redisService.setLTS(longUrl, shortUrl, time);
+        System.out.println("inlong");
+        redisService.setSTL(longUrl, shortUrl, time);
+        System.out.println("inshort");
```

Comment:
should use Redis persistence to save  counter

Reply:
not sure what you mean, i have saved ID into redis in sequenceIdService.java

Reply:
Please talk to @zhangxyleon to get this comments resolved